### PR TITLE
harfbuzz 4.4.1

### PR DESCRIPTION
**Pull request by triallax**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
Linux probably needs a newer compiler. Or different `CFLAGS`. @Homebrew/linux

**Comment by cho-m** [comment]
Fixed upstream with https://github.com/harfbuzz/harfbuzz/pull/3707, but the commit doesn't apply cleanly.

PR links what changes are needed on 4.4.1: https://github.com/harfbuzz/harfbuzz/compare/4.4.1...kleisauke:harfbuzz:workaround-gcc-bug-4.4.1

File: Formula/harfbuzz.rb [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
-  sha256 "32184860ddc0b264ff95010e1c64e596bd746fe4c2e34014a1185340cdddeba6"
   license "MIT"
   head "https://github.com/harfbuzz/harfbuzz.git", branch: "main"
 
+  stable do
+    url "https://github.com/harfbuzz/harfbuzz/archive/4.4.1.tar.gz"
+    sha256 "1a95b091a40546a211b6f38a65ccd0950fa5be38d95c77b5c4fa245130b418e1"
+
+    # Fix build on GCC < 7, remove on next release.
+    patch :DATA
```
Comment (carlocab) [inline Formula/harfbuzz.rb:12]: Can you submit this patch to https://github.com/Homebrew/formula-patches?

Comment (carlocab) [inline Formula/harfbuzz.rb:12]:
```suggestion
    patch do
      url "https://raw.githubusercontent.com/Homebrew/formula-patches/ae5613e951257f508f4b17e9e24a3ea2ccb43a3f/harfbuzz/fix-pregcc7-build.patch"
      sha256 "17abbae47e09a0daa3f5afa5f6ba37353db00c2f0fe025a014856d8b023672b6"
    end
```

File: Formula/harfbuzz.rb [hunk 2]
```diff
@@ -56,3 +62,200 @@ def install
...
+@@ -82,7 +81,8 @@ inline bool SubstLookup::dispatch_recurse_func<hb_ot_apply_context_t> (hb_ot_app
+ }
+ #endif
+ 
+-
++} /* namespace GSUB */
++} /* namespace Layout */
+ } /* namespace OT */
+ 
+
```
Comment (carlocab) [inline Formula/harfbuzz.rb:261]:
```suggestion
```

Comment (carlocab) [inline Formula/harfbuzz.rb:261]:
```suggestion
```

File: Formula/harfbuzz.rb [hunk 3]
```diff
@@ -56,3 +62,200 @@ def install
     end
   end
 end
+__END__
```
[the remark this answers is not in the document]
Comment (mhmdanas) [reply Formula/harfbuzz.rb]: I see, thanks for pointing this out.

**Review by SMillerDev — DISMISSED** [review]

**Review by carlocab — APPROVED** [review]

**Comment by SMillerDev** [comment]
Did you mean to get rid of `long build` too?

**Comment by carlocab** [comment]
I did. Just forgot 😅

**Pull request closed without merging**